File: PoZiomkaDomain/Match/Dtos/MatchModel.cs
namespace PoZiomkaDomain.Match.Dtos;

public enum MatchStatus
{
    Accepted,
    Rejected
}

public record MatchModel(int Id, int StudentId1, int StudentId2, MatchStatus Status1, MatchStatus Status2)
{
    public bool IsAccepted { get => Status1 == MatchStatus.Accepted && Status2 == MatchStatus.Accepted; }
    public bool IsRejected { get => Status1 == MatchStatus.Rejected || Status2 == MatchStatus.Rejected; }
};

File: PoZiomkaApp/src/app/student/matches/match.model.ts
export enum MatchStatus {
  pending, accepted, rejected
};

export type MatchModel = {
  id: number,
  studentId1: number,
  studentId2: number,
  status1: MatchStatus,
  status2: MatchStatus
  isAccepted: boolean,
  isRejected: boolean
};
